src/Bicep.Core/Semantics/ArmTemplateSemanticModel.cs

```diff
@@ -391,29 +391,19 @@ private TypeSymbol GetObjectType(Template template, ITemplateSchemaNode schemaNo
             _ => GetType(schemaNode) switch { TypeSymbol concreteType => (concreteType, concreteType.Name) },
         };
 
-        private static TypeSymbol GetType(TemplateOutputParameter output)
+        private TypeSymbol GetType(TemplateOutputParameter output)
         {
-            return output.Type.Value switch
+            var resolved = TemplateEngine.ResolveSchemaReferences(SourceFile.Template, output);
+
+            return resolved.Type.Value switch
```

Comment:
This will work (because a schema node in an ARM template must have either a `$ref` or a `type` property, so the value returned from `TemplateEngine.ResolveSchemaReferences` will always have a `.Type` property), but the references will get resolved anew in `GetType(ITemplateSchemaNode)`. It would be slightly more efficient to instead just treat `output.Type` as nullable:
```suggestion
            return output.Type?.Value switch
```

In a follow-up PR, I'm going to see if this class can be simplified a little bit. There are specialized `GetType` methods for parameters and outputs, but I'm not sure that's necessary any more.

Reply:
i broke my test cases, trying to figure out which change it was.